From 26c4d9e10efdbaf613c623ebf94ecb940de64db5 Mon Sep 17 00:00:00 2001
From: Yifan Hong <elsk@google.com>
Date: Mon, 9 Oct 2017 15:21:15 -0700
Subject: [PATCH 118/296] Fix BatteryManager.getIntProperty returns 0 as
 invalid value.

Long.MIN_VALUE cast to Integer.MIN_VALUE always returns 0.
---
 core/java/android/os/BatteryManager.java | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/core/java/android/os/BatteryManager.java b/core/java/android/os/BatteryManager.java
index f715f507f6c..9eccb375fc4 100644
--- a/core/java/android/os/BatteryManager.java
+++ b/core/java/android/os/BatteryManager.java
@@ -287,7 +287,8 @@ public class BatteryManager {
      * @return the property value, or Integer.MIN_VALUE if not supported.
      */
     public int getIntProperty(int id) {
-        return (int)queryProperty(id);
+        long val = queryProperty(id);
+        return val == Long.MIN_VALUE ? Integer.MIN_VALUE : (int)val;
     }
 
     /**
-- 
2.11.0

